cmd/vela-terraform/plugin_test.go

```diff
@@ -4,4 +4,266 @@
 
 package main
 
-// TODO: revisit this file
+import (
+	"testing"
+	"time"
+)
+
+func TestKubernetes_Plugin_Validate(t *testing.T) {
+	// setup tests
+	tests := []struct {
+		plugin *Plugin
+		want   *error
+	}{
+		{ // test success for apply action
+			plugin: &Plugin{
+				Apply: &Apply{
+					AutoApprove: true,
+					Backup:      "backup/",
+					Directory:   "foobar/",
+					Lock:        true,
+					LockTimeout: 1 * time.Second,
+					Input:       true,
+					NoColor:     true,
+					Parallelism: 1,
+					Refresh:     true,
+					State:       "state.tf",
+					StateOut:    "stateout.tf",
+					Target:      "target.tf",
+					Vars:        []string{"foo=bar", "bar=foo"},
+					VarFiles:    []string{"vars1.tf", "vars2.tf"},
+				},
+				Config: &Config{
+					Action: "apply",
+					Netrc: &Netrc{
+						Machine:  "machine.company.com",
```

Comment:
```suggestion
						Machine:  "machine.example.com",
```
i know it's just a test, but probably better not to use a real company's domain